src/Database/Query.php

```diff
@@ -179,7 +179,7 @@ public function getConnection()
      *
      * @deprecated 3.4.0 Use setConnection()/getConnection() instead.
      * @param \Cake\Database\Connection|null $connection Connection instance
-     * @return $this|\Cake\Database\Connection
+     * @return static|\Cake\Database\Connection
```

Comment:
Wasn't there a reason (IDEs) that we stuck to the $this for fluent interface and non static?

Reply:
I can fix the code in Psalm that demands this change (only `@return $this` is currently permitted). No good reason, though I do prefer `static` a little.

Reply:
Afaik static broke the fluent interface approach in IDEs. We would have to test if this change is good.
The deprecated methods should not be in use anymore though, so it most likely won't be too big of a deal either way. And for future methods we sure always try to only use true fluent typehinting (no getter mixin).

Reply:
But I do see a few methods that are not even deprecated yet, they should be separated into their getter/setter part first IMO.

Reply:
it would surprise me if IDEs (i.e. PHPStorm) didn't support `@return static` at this point.

Reply:
I feel like this changes back and forth every other month